__tests__/wallet/wallet.test.ts

```diff
@@ -381,6 +383,238 @@ test('getTxById', async () => {
+  await expect(wallet.prepareMintTokensData('01', 100, {
+    address: addresses[1],
+    createAnotherMint: true,
+    mintAuthorityAddress: 'abc',
+    allowExternalMintAuthorityAddress: true,
+    pinCode: '123456',
+  })).rejects.toThrowError(SendTxError);
+
+  // mint data with correct address for authority output
+  const mintData = await wallet.prepareMintTokensData('01', 100, {
+    address: addresses[1],
+    createAnotherMint: true,
+    mintAuthorityAddress: addresses[2],
+    pinCode: '123456',
+  });
+
+  expect(mintData.outputs).toHaveLength(2);
+
+  const authorityOutputs = mintData.outputs.filter(
+    o => transaction.isAuthorityOutput({ token_data: o.tokenData })
+  );
+
+  expect(authorityOutputs).toHaveLength(1);
+  const authorityOutput = authorityOutputs[0];
+  expect(authorityOutput.value).toEqual(TOKEN_MINT_MASK);
+  const p2pkh = authorityOutput.parseScript(network);
+  // Validate that the authority output was sent to the correct address
+  expect(p2pkh.address.base58).toEqual(addresses[2]);
+
+  // Clear mocks
+  spy1.mockRestore();
+  spy2.mockRestore();
+  spy3.mockRestore();
+});
+
+test('prepareMeltTokens', async () => {
+  const addresses = [
+    'WdSD7aytFEZ5Hp8quhqu3wUCsyyGqcneMu',
+    'WbjNdAGBWAkCS2QVpqmacKXNy8WVXatXNM',
+    'WR1i8USJWQuaU423fwuFQbezfevmT4vFWX',
+  ]
+
+  mockAxiosAdapter.onPost('wallet/addresses/check_mine').reply(200, {
+    success: true,
+    addresses: {
+      'WR1i8USJWQuaU423fwuFQbezfevmT4vFWX': true,
+    },
+  });
+
+  const requestPassword = jest.fn();
+  const network = new Network('testnet');
+  const seed = 'purse orchard camera cloud piece joke hospital mechanic timber horror shoulder rebuild you decrease garlic derive rebuild random naive elbow depart okay parrot cliff';
+  const wallet = new HathorWalletServiceWallet({
+    requestPassword,
+    seed,
+    network,
+    passphrase: '',
+    xpriv: null,
+    xpub: null,
+  });
+
+  const code = new Mnemonic(seed);
+  const xpriv = code.toHDPrivateKey('', network.getNetwork());
+
+  wallet.setState('Ready');
+
+  const getUtxosMock = async (params) => {
+    if (params.authority === TOKEN_MELT_MASK) {
+      return {
+        utxos: [{
+          txId: '002abde4018935e1bbde9600ef79c637adf42385fb1816ec284d702b7bb9ef5f',
+          index: 0,
+          tokenId: '01',
+          address: addresses[0],
+          value: 1,
+          authorities: TOKEN_MELT_MASK,
+          timelock: null,
+          heightlock: null,
+          locked: false,
+          addressPath: 'm/280\'/280\'/0/1/0',
+        }],
+        changeAmount: 0,
+      };
+    } else {
+      return {
+        utxos: [{
+          txId: '002abde4018935e1bbde9600ef79c637adf42385fb1816ec284d702b7bb9ef5d',
+          index: 0,
+          tokenId: '01',
+          address: addresses[0],
+          value: 1,
+          authorities: 0,
+          timelock: null,
+          heightlock: null,
+          locked: false,
+          addressPath: 'm/280\'/280\'/0/1/0',
```

Comment:
Use a constant for this addressPath, it's being repeated all over this file

Reply:
Done https://github.com/HathorNetwork/hathor-wallet-lib/pull/519/commits/435658f40cac1aea5e04372c614b3fcbacd84c4d